### PR TITLE
chore: Support claude review

### PR DESCRIPTION
**Pull request by Sneh1999**
# Description
Add claude review to our repo

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1213818023113844

### CODE REVIEW

File: .github/workflows/claude-review.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+               - Review database queries for efficiency
+               - Check for memory leaks or resource issues
+
+            4. **Testing**
+               - Verify adequate test coverage
+               - Review test quality and edge cases
+               - Check for missing test scenarios
+
+            5. **Documentation**
+               - Ensure code is properly documented
```
Comment (jjeangal) [inline .github/workflows/claude-review.yml:70]: nit: Do we maybe want to ensure that it should not be over documented to?

**Comment by Sneh1999** [comment]
@claude

**Comment by sveitser** [comment]
@claude review

**Comment by Sneh1999** [comment]
@claude review

**Comment by Sneh1999** [comment]
@claude review

File: .github/workflows/claude.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      actions: read
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - uses: anthropics/claude-code-action@v1
+        with:
+          anthropic_api_key: ${{ secrets.ANTHROPIC_API_KEY }}
+          claude_args: "--model claude-opus-4-6 --max-turns 15"
```
Comment (jjeangal) [inline .github/workflows/claude.yml:39]: Can we let's say choose different models for different tasks?

**Review by jjeangal — COMMENTED** [review]
A few nits

**Comment by zacshowa** [comment]
@claude review

**Comment by zacshowa** [comment]
@Sneh1999 It looks like the claude review didn't actually complete a review because this PR is the one introducing the workflow, and the workflow won't work on non default branches until it exists in the default branch.
Otherwise, it was successfully able to get the github token from secrets, so this should work in the future. The only thing that seemed to trigger the job was setting this PR away from ready to review, i.e. making it a draft, and then marking it as ready to review again.

**Comment by Sneh1999** [comment]
@zacshowa @jjeangal  this code is from Mathis so we want to merge it as is, Mathis did say it wont work until we merge the PR so just need a approval so that we can test it out

**Review by zacshowa — APPROVED** [review]
